app/scripts/controllers/app-state-controller.ts

```diff
@@ -0,0 +1,860 @@
+
+export type AppStateControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  AppStateControllerActions | AllowedActions,
+  AppStateControllerEvents | AllowedEvents,
+  AllowedActions['type'],
+  AllowedEvents['type']
+>;
+
+type PollingTokenType =
+  | 'popupGasPollTokens'
+  | 'notificationGasPollTokens'
+  | 'fullScreenGasPollTokens';
+
+type SecurityAlertResponse = {
+  block?: number;
+  description?: string;
+  features?: string[];
+  providerRequestsCount?: Record<string, number>;
+  reason: string;
+  result_type: string;
+  securityAlertId?: string;
+  source?: SecurityAlertSource;
+};
+
+type AppStateControllerOptions = {
+  addUnlockListener: (callback: () => void) => void;
+  isUnlocked: () => boolean;
+  initState?: Partial<AppStateControllerState>;
+  onInactiveTimeout?: () => void;
+  // TODO: Remove this as soon as PreferencesController upgrade to BaseControllerV2 merges with develop
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  preferencesStore: any;
+  messenger: AppStateControllerMessenger;
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  extension: any;
+};
+
+const getInitState = (
+  initState?: Partial<AppStateControllerState>,
+): AppStateControllerState => ({
+  timeoutMinutes: DEFAULT_AUTO_LOCK_TIME_LIMIT,
+  connectedStatusPopoverHasBeenShown: true,
+  defaultHomeActiveTabName: null,
+  browserEnvironment: {},
+  popupGasPollTokens: [],
+  notificationGasPollTokens: [],
+  fullScreenGasPollTokens: [],
+  recoveryPhraseReminderHasBeenShown: false,
+  recoveryPhraseReminderLastShown: new Date().getTime(),
+  outdatedBrowserWarningLastShown: null,
+  nftsDetectionNoticeDismissed: false,
+  showTestnetMessageInDropdown: true,
+  showBetaHeader: isBeta(),
+  showPermissionsTour: true,
+  showNetworkBanner: true,
+  showAccountBanner: true,
+  trezorModel: null,
+  onboardingDate: null,
+  newPrivacyPolicyToastClickedOrClosed: null,
+  newPrivacyPolicyToastShownDate: null,
+  hadAdvancedGasFeesSetPriorToMigration92_3: false,
+  ...initState,
+  qrHardware: {},
+  nftsDropdownState: {},
+  usedNetworks: {
+    '0x1': true,
+    '0x5': true,
+    '0x539': true,
+  },
+  surveyLinkLastClickedOrClosed: null,
+  signatureSecurityAlertResponses: {},
+  switchedNetworkDetails: null,
+  switchedNetworkNeverShowMessage: false,
+  currentExtensionPopupId: 0,
+});
+
+export default class AppStateController extends EventEmitter {
+  // TODO: Replace `any` with type
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private extension: any;
+
+  private onInactiveTimeout: () => void;
+
+  store: ObservableStore<AppStateControllerState>;
+
+  private timer: NodeJS.Timeout | null;
+
+  private isUnlocked: () => boolean;
+
+  private waitingForUnlock: { resolve: () => void }[];
+
+  private messagingSystem: AppStateControllerMessenger;
+
+  private _approvalRequestId: string | null;
```

Comment:
can we use the `#` pattern for private variables... instead of `_` or it's planned for BaseController upgrade ?

Reply:
I feel like this is more related to the typescript conversion than the BaseController upgrade. Though I agree that `#` variables should be preferred